src/rgw/rgw_sal_motr.h

```diff
@@ -501,6 +501,27 @@ struct AccumulateIOCtxt{
   uint64_t total_bufer_sz = 0;
 };
 
+class MotrCopyObj_Filter : public RGWGetDataCB {
+public:
+  virtual int handle_data(bufferlist& bl, off_t bl_ofs, off_t bl_len) override { return 0; }
+  MotrCopyObj_Filter() {}
+  virtual ~MotrCopyObj_Filter() override {}
+};
+
+class MotrCopyObj_CB : public MotrCopyObj_Filter
+{
+  const DoutPrefixProvider* m_dpp;
+  std::shared_ptr<rgw::sal::Writer> m_dst_writer;
```

Comment:
Add one more variable as `uint64 offset` to track the data written to object so far.

Reply:
Made some changes @sjain09. Please review